### ISSUE TITLE
LT01 false positives when using BigQuery concatenation operator

### ISSUE THREAD

**Issue by tver-higashi** [sqlfluff/sqlfluff#5817]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

sqlfluff does not correctly recognize BigQuery's string concatenation operator (`||`) as a binary operator.

### Expected Behaviour

The "sqlfluff lint" command succeeds without an error.

### Observed Behaviour

The command error with this output:

```
# echo 'SELECT "a" || "b" AS s FROM t;' | sqlfluff lint --config .sqlfluff
== [/opt/env/devintegrate/hoge.sql] FAIL                                                                                                                                                                    
L:   1 | P:  13 | LT01 | Expected single whitespace between pipe '|' and pipe
                       | '|'. [layout.spacing]
```

### How to reproduce

Run `echo 'SELECT "a" || "b" AS s FROM t;' | sqlfluff lint --config .sqlfluff`

In BigQuery, `||` is the string concatenation operator.
But this has been misidentified.
https://cloud.google.com/bigquery/docs/reference/standard-sql/operators

### Dialect

bigquery

### Version

3.0.5

### Configuration

[sqlfluff]
dialect = bigquery
rules = core

[sqlfluff:layout:type:binary_operator]
spacing_within = single
line_position = leading

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
